session.go

```diff
@@ -1431,14 +1467,16 @@ func (b *Batch) Keyspace() string {
 }
 
 // Attempts returns the number of attempts made to execute the batch.
-func (b *Batch) Attempts() int {
-	return b.attempts
+func (b *Batch) Attempts(host *HostInfo) uint64 {
+	hostMetric := b.getHostMetrics(host)
+	return hostMetric.attempts
 }
 
 //Latency returns the average number of nanoseconds to execute a single attempt of the batch.
-func (b *Batch) Latency() int64 {
-	if b.attempts > 0 {
-		return b.totalLatency / int64(b.attempts)
+func (b *Batch) Latency(host *HostInfo) int64 {
+	hostMetric := b.getHostMetrics(host)
```

Comment:
Why doesn't the `Batch` have a mutex while the `Query` still does? 🤔

Reply:
They both don't have a mutex. The only mutexes in Query (but also in Batches) are the ones embedded in Session, where there are 2: one for the session itself and the second is for the hostPool access. Am I missing something? :-)

Reply:
Don't know what I was seeing – weird. Diregard. 😬